app/src/main/java/com/aaronjwood/portauthority/network/Wireless.java

```diff
@@ -152,11 +154,30 @@ public String getInternalWifiIpAddress() {
      * @return Internal Wifi Subnet Netmask
      */
     public int getInternalWifiSubnet() {
-        // ToDo: Make sure this works with static DHCP reservations.
-        //FIXME this seems to have issues on some phones running 5.0+ https://code.google.com/p/android/issues/detail?id=82477
         WifiManager wifiManager = this.getWifiManager();
         DhcpInfo dhcpInfo = wifiManager.getDhcpInfo();
-        return dhcpInfo.netmask; //FIXME returns 0
+        int netmask = Integer.bitCount(dhcpInfo.netmask);
+        /*
+         * Workaround for #82477
+         * https://code.google.com/p/android/issues/detail?id=82477
+         * If dhcpInfo returns a subnet that cannot exist, then
+         * look up the Network interface instead.
+         */
+        if (dhcpInfo.netmask < 8 || dhcpInfo.netmask > 32) {
```

Comment:
31^2 hosts would be insane to scan. I don't know anyone who is using < 8.  Regardless, it's trivial to support, so sure!